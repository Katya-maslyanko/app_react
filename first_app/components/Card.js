// Импортируем необходимые модули и компоненты
import React from "react";
import { useNavigation } from "@react-navigation/native";
import { View, Image } from "react-native";
import { COLORS, SIZES, SHADOWS } from "../constants";
import { SubInfo, Title } from "./SubInfo";
import { RectButton } from "./Button";

const Card = ({ data }) => {
  // Получаем объект навигации из React Navigation
  const navigation = useNavigation();

  return (
    <View
      style={{
        backgroundColor: COLORS.white,
        borderRadius: SIZES.font,
        marginBottom: SIZES.extraLarge,
        margin: SIZES.base,
        ...SHADOWS.dark,
      }}
    >
      {/* Контейнер для изображения */}
      <View
        style={{
          width: "100%",
          height: 100,
        }}
      >
        <Image
          source={data.image}
          resizeMode="cover"
          style={{
            width: "100%",
            height: "100%",
            borderTopLeftRadius: SIZES.font,
            borderTopRightRadius: SIZES.font,
          }}
        />
      </View>

      {/* Компонент для отображения дополнительной информации */}
      <SubInfo />

      <View style={{ width: "100%", padding: SIZES.font }}>
        {/* Компонент для отображения названия и создателя  */}
        <Title
          title={data.name}
          subTitle={data.creator}
          titleSize={SIZES.large}
          subTitleSize={SIZES.small}
        />

        <View
          style={{
            marginTop: SIZES.font,
            flexDirection: "row",
            justifyContent: "space-between",
            alignItems: "center",
            justifyContent: "flex-end",
          }}
        >
          {/* Кнопка для перехода на экран деталей  */}
          <RectButton
            minWidth={120}
            fontSize={SIZES.font}
            handlePress={() => navigation.navigate("Details", { data })}
          />
        </View>
      </View>
    </View>
  );
};

export default Card;