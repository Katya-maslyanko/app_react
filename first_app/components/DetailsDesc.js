import React, { useState } from "react";
import { View, Text } from "react-native";
import { Title } from "./SubInfo"; // Импортируем компонент Title из файла SubInfo
import { COLORS, SIZES, FONTS } from "../constants"; // Импортируем константы из файла constants

const DetailsDesc = ({ data }) => {
  // Используем хук useState для управления состоянием текста описания и флага readMore
  const [text, setText] = useState(data.description.slice(0, 100)); // Изначально отображаем первые 100 символов описания
  const [readMore, setReadMore] = useState(false); // Изначально флаг readMore установлен в false

  return (
    <>
      <View
        style={{
          width: "100%",
          flexDirection: "row",
          justifyContent: "space-between",
          alignItems: "center",
        }}
      >
        {/* Отображаем компонент Title с данными из props */}
        <Title
          title={data.name}
          subTitle={data.creator}
          titleSize={SIZES.extraLarge}
          subTitleSize={SIZES.font}
        />
      </View>

      <View style={{ marginVertical: SIZES.extraLarge * 1.5 }}>
        <Text
          style={{
            fontSize: SIZES.font,
            fontFamily: FONTS.semiBold,
            color: COLORS.primary,
          }}
        >
          Описание поста:
        </Text>
        <View
          style={{
            marginTop: SIZES.base,
          }}
        >
          <Text
            style={{
              color: COLORS.secondary,
              fontSize: SIZES.small,
              fontFamily: FONTS.regular,
              lineHeight: SIZES.large,
            }}
          >
            {/* Отображаем текст описания */}
            {text}
            {/* Если readMore не установлен, отображаем многоточие (...) */}
            {!readMore && "..."}
          </Text>
        </View>
      </View>
    </>
  );
};

export default DetailsDesc;