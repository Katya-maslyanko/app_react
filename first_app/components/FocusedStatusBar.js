// Импортируем необходимые модули и компоненты
import React from "react";
import { StatusBar } from "react-native";
import { useIsFocused } from "@react-navigation/core";

const FocusedStatusBar = (props) => {
  // Используем хук useIsFocused из @react-navigation/core
  // для определения, является ли текущий экран активным (фокусированным)
  const isFocused = useIsFocused();

  // Если экран фокусирован, отображаем StatusBar с переданными свойствами и анимацией
  // В противном случае не отображаем StatusBar
  return isFocused ? <StatusBar animated={true} {...props} /> : null;
};

export default FocusedStatusBar;