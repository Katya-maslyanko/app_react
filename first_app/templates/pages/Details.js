import React from "react";
import { View, Text, SafeAreaView, Image, StatusBar, FlatList,  Platform, StyleSheet} from "react-native";
import { COLORS, SIZES, assets, FONTS } from "../../constants";
import { CircleButton, SubInfo, DetailsDesc, FocusedStatusBar } from "../../components";
import { useNavigation } from "@react-navigation/native";
// Импортируем необходимые модули
import Constants from 'expo-constants';

// Компонент заголовка экрана деталей и кнопкк назад в Home
const DetailsHeader = ({ data, navigation }) => {
  // Определяем стили в зависимости от платформы
  const styles = Platform.select({
    ios: {
      container: {
        width: "100%",
        height: 420,
      },
      image: {
        width: "100%",
        height: "100%",
      },
    },
    android: {
      container: {
        width: "100%",
        height: 373,
      },
      image: {
        width: "100%",
        height: "100%",
      },
    },
  });

  return (
    <View style={styles.container}>
      <Image source={data.image} resizeMode="cover" style={styles.image} />

      <CircleButton
        imgUrl={assets.left}
        handlePress={() => navigation.navigate("Home")}
        left={15}
        top={StatusBar.currentHeight + 10}
      />
    </View>
  );
};

// Компонент, показывающий платформу
const AppPlatform = () => {
  return (
    <View style={styles.platformContainer}>
      <Text style={styles.platformText}>
        {Platform.OS === 'ios' ? 'iOS' : 'Android'}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  platformContainer: {
    position: 'absolute',
    bottom: 20,
    right: 20,
    backgroundColor: COLORS.white,
    paddingVertical: SIZES.small,
    paddingHorizontal: SIZES.font,
    borderRadius: SIZES.font,
    shadowColor: COLORS.gray,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  platformText: {
    fontFamily: FONTS.regular,
    fontSize: SIZES.small,
    color: COLORS.primary,
  },
});


// Главный компонент экрана деталей
const Details = ({ route, navigation }) => {
  const { data } = route.params;
  const styles = Platform.select({
    ios: StyleSheet.create({
      container: {
        flex: 1,
      },
      contentContainer: {
        paddingBottom: SIZES.extraLarge * 4, // Больший отступ снизу для iOS
      },
    }),
    android: StyleSheet.create({
      container: {
        flex: 1,
      },
      contentContainer: {
        paddingBottom: SIZES.extraLarge * 3, // Меньший отступ снизу для Android
      },
    }),
  });


  return (
    <SafeAreaView style={styles.container}>
      {/* Компонент строки состояния, адаптированной для навигации */}
      <FocusedStatusBar
        barStyle="dark-content"
        backgroundColor="transparent"
        translucent={true}
      />

      {/* FlatList для отображения списка элементов */}
      <FlatList
        data={data.bids}
        keyExtractor={(item) => item.id}
        showsVerticalScrollIndicator={false}
        contentContainerStyle={styles.contentContainer}
        ListHeaderComponent={() => (
          <React.Fragment>
            {/* Вызов компонента заголовка */}
            <DetailsHeader data={data} navigation={navigation} />
            {/* Вызов компонента дополнительной информации */}
            <SubInfo />
            <View style={{ padding: SIZES.font }}>
              {/* Вызов компонента описания деталей */}
              <DetailsDesc data={data} />
            </View>
          </React.Fragment>
        )}
      />
    <AppPlatform />
    </SafeAreaView>
  );
};


export { DetailsHeader, AppPlatform };
export default Details;