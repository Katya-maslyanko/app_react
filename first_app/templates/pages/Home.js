// Импортируем необходимые модули и компоненты
import React, { useState } from "react";
import { View, SafeAreaView, FlatList, Platform, TouchableOpacity } from "react-native";
import { Card, HomeHeader, FocusedStatusBar } from "../../components";
import { COLORS, PostData, SIZES, SHADOWS } from "../../constants";
  
const Home = () => {
    // Определяем состояние PostData и функцию sePostData для управления данными Post
    const [searchText, setSearchText] = useState("");
    const [filteredPosts, setFilteredPosts] = useState(PostData);    

    // Функция обработки поиска
    const handleSearch = (value) => {
        setSearchText(value);
        const filtered = PostData.filter((post) =>
          post.name && post.name.toLowerCase().includes(value.toLowerCase())
        );
        setFilteredPosts(filtered);
      };               

    // Определяем стили в зависимости от платформы
    const styles = Platform.select({
        ios: {
        container: {
            flex: 1,
            backgroundColor: COLORS.primary,
        },
        headerContainer: {
            zIndex: 0,
            backgroundColor: COLORS.primary,
        },
        },
        android: {
        container: {
            flex: 1,
            backgroundColor: COLORS.secondary,
        },
        headerContainer: {
            zIndex: 0,
            backgroundColor: COLORS.secondary,
        },
        },
    });

    return (
        <SafeAreaView style={styles.container}>
            {/* Настраиваем цвет строки состояния */}
            <FocusedStatusBar backgroundColor={COLORS.primary} />
            <View style={styles.container}>
                <View style={styles.headerContainer}>
                    {/* Отображаем списком данные Post с помощью FlatList */}
                    <FlatList
                    data={filteredPosts}
                    renderItem={({ item }) => <Card data={item} />}
                    keyExtractor={(item) => item.id}
                    showsVerticalScrollIndicator={false}
                    ListHeaderComponent={<HomeHeader onSearch={handleSearch} value={searchText} />}
                    />
                </View>

                {/* Этот View используется для создания эффекта наложения */}
                <View
                    style={{
                        position: "absolute",
                        top: 0,
                        bottom: 0,
                        right: 0,
                        left: 0,
                        zIndex: -1,
                    }}
                />
            </View>
        </SafeAreaView>
    );
};

export default Home;