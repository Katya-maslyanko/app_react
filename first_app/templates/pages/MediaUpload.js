import React, { useState } from "react";
import { View, Text, TouchableOpacity, Image, StyleSheet, FlatList, StatusBar } from "react-native";
import { useNavigation } from "@react-navigation/native";
import * as ImagePicker from "expo-image-picker";
import { Video } from "react-native-video";
import { COLORS, SIZES, FONTS, assets } from "../../constants";
import { CircleButton } from "../../components/Button";

const MediaUpload = () => {
  // Получаем объект навигации с помощью хука useNavigation
  const navigation = useNavigation();
  // Состояние выбранных медиа-файлов и флаг, определяющий, содержит ли выбранное медиа видео
  const [selectedMedia, setSelectedMedia] = useState([]);
  const [isVideo, setIsVideo] = useState(false);
  // Функция, обрабатывающая выбор медиа-файлов
  const handleMediaPicker = async () => {
    // Открываем библиотеку изображений и разрешаем множественный выбор
    const result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsMultipleSelection: true,
    });

    // Если пользователь не отменил выбор и выбрал какие-либо файлы
    if (!result.cancelled && result.assets) {
      // Сохраняем выбранные медиа-файлы и обновляем состояние isVideo
      setSelectedMedia(result.assets);
      setIsVideo(result.assets.some((item) => item.type === "video"));
    }
  };

  // Функция, обрабатывающая нажатие на кнопку "Отобразить"
  const handleShowMedia = () => {
    // Если есть выбранные медиа-файлы
    if (selectedMedia?.length > 0) {
      // Обновляем состояние isVideo в зависимости от выбранных медиа
      setIsVideo(selectedMedia.some((item) => item.type === "video"));
      // Сохраняем выбранные медиа-файлы
      setSelectedMedia(selectedMedia);
    }
  };

  // Функция для отрисовки каждого медиа-элемента
  const renderMediaItem = ({ item }) => {
    try {
      // Отрисовываем видео или изображение в зависимости от типа элемента
      return (
        <View style={styles.mediaItem}>
          {item.type === "video" ? (
            <Video
              source={{ uri: item.uri }}
              style={styles.videoContainer}
              resizeMode="contain"
              useNativeControls
            />
          ) : (
            <Image source={{ uri: item.uri }} style={styles.media} />
          )}
        </View>
      );
    } catch (error) {
      // Логируем любые ошибки, возникающие при отрисовке
      console.error("Error rendering media item:", error);
      return null;
    }
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        {/* Отрисовываем кнопку "Назад" в хедере */}
        <CircleButton
          imgUrl={assets.left}
          handlePress={() => navigation.navigate("Home")}
          left={15}
          top={StatusBar.currentHeight + 10}
        />
      </View>
      <View style={styles.addMediaContainer}>
        {/* Отрисовываем кнопку "Загрузить медиа" */}
        <TouchableOpacity style={styles.addMediaButton} onPress={handleMediaPicker}>
          <Text style={styles.addMediaButtonText}>Загрузить медиа</Text>
        </TouchableOpacity>
        {/* Отрисовываем кнопку "Отобразить" */}
        <TouchableOpacity style={styles.showMediaButton} onPress={handleShowMedia}>
          <Text style={styles.showMediaButtonText}>Отобразить</Text>
        </TouchableOpacity>
      </View>
      {/* Если есть выбранные медиа-файлы, отрисовываем их в виде сетки */}
      {selectedMedia && selectedMedia.length > 0 && (
        <View style={styles.mediaContainer}>
          <FlatList
            data={selectedMedia}
            keyExtractor={(item) => item.uri}
            numColumns={3}
            renderItem={renderMediaItem}
            contentContainerStyle={styles.mediaGrid}
          />
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.primary,
    padding: SIZES.font,
  },
  header: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "flex-start",
    marginBottom: SIZES.font,
    marginTop: 20,
  },
  addMediaContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    backgroundColor: COLORS.gray,
    borderRadius: SIZES.font,
    paddingVertical: SIZES.small,
    paddingHorizontal: SIZES.font,
    marginBottom: SIZES.font,
    marginTop: 50,
  },
  addMediaButton: {
    backgroundColor: COLORS.gray,
    borderRadius: SIZES.font,
    paddingVertical: SIZES.small,
    paddingHorizontal: SIZES.font,
  },
  addMediaButtonText: {
    fontFamily: FONTS.bold,
    fontSize: SIZES.medium,
    color: COLORS.white,
  },
  showMediaButton: {
    backgroundColor: COLORS.primary,
    borderRadius: SIZES.font,
    paddingVertical: SIZES.small,
    paddingHorizontal: SIZES.font,
  },
  showMediaButtonText: {
    fontFamily: FONTS.bold,
    fontSize: SIZES.medium,
    color: COLORS.white,
  },
  mediaContainer: {
    flex: 1,
    borderRadius: SIZES.font,
    justifyContent: "center",
    alignItems: "center",
  },
  mediaGrid: {
    padding: SIZES.font,
  },
  mediaItem: {
    width: "33.33%",
    height: 100,
    padding: SIZES.small,
  },
  media: {
    width: "100%",
    height: "100%",
  },
  videoContainer: {
    width: "100%",
    height: 200,
  },
});

export default MediaUpload;