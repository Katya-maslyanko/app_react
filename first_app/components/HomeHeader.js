// –ò–º–ø–æ—Ä—Ç–∏—Ä—É–µ–º –Ω–µ–æ–±—Ö–æ–¥–∏–º—ã–µ –º–æ–¥—É–ª–∏ –∏ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç—ã
import React from "react";
import { View, Text, Image, TextInput } from "react-native";
import { COLORS, FONTS, SIZES, assets } from "../constants";
import { useNavigation } from "@react-navigation/native";
import { TouchableOpacity, StyleSheet } from "react-native";
import MediaUpload from "../templates/pages/MediaUpload";

const HomeHeader = ({ onSearch }) => {
  const navigation = useNavigation();

  return (
    <View
      style={{
        backgroundColor: COLORS.primary,
        padding: SIZES.font,
      }}
    >
      {/* –ö–æ–Ω—Ç–µ–π–Ω–µ—Ä –¥–ª—è –≤–µ—Ä—Ö–Ω–µ–π —á–∞—Å—Ç–∏ –∑–∞–≥–æ–ª–æ–≤–∫–∞ */}
      <View
        style={{
          flexDirection: "row",
          justifyContent: "space-between",
          alignItems: "center",
        }}
      >
        {/* –ö–æ–Ω—Ç–µ–π–Ω–µ—Ä –¥–ª—è –ø—Ä–∏–≤–µ—Ç—Å—Ç–≤–∏—è –∏ –∑–∞–≥–æ–ª–æ–≤–∫–∞ */}
        <View style={{ marginVertical: SIZES.font }}>
          <Text
            style={{
              fontFamily: FONTS.regular,
              fontSize: SIZES.small,
              color: COLORS.white,
            }}
          >
            –ü—Ä–∏–≤–µ—Ç, –ï–∫–∞—Ç–µ—Ä–∏–Ω–∞ üëã
          </Text>

          <Text
            style={{
              fontFamily: FONTS.bold,
              fontSize: SIZES.medium,
              color: COLORS.white,
              marginTop: SIZES.base / 2,
            }}
          >
            –ü—Ä–æ—Å–º–æ—Ç—Ä –ø–æ—Å—Ç–æ–≤
          </Text>
        </View>
        <TouchableOpacity
        style={{
          backgroundColor: COLORS.gray,
          padding: SIZES.small,
          borderRadius: SIZES.font,
          alignItems: "center",
          marginTop: SIZES.base,
        }}
        onPress={() => navigation.navigate("MediaUpload")}
      >
        <Text style={{ fontFamily: FONTS.bold, color: COLORS.white }}>
          –ì–∞–ª–µ—Ä–µ—è –º–µ–¥–∏–∞
        </Text>
      </TouchableOpacity>

        {/* –ö–æ–Ω—Ç–µ–π–Ω–µ—Ä –¥–ª—è –∞–≤–∞—Ç–∞—Ä–∞ –ø–æ–ª—å–∑–æ–≤–∞—Ç–µ–ª—è */}
        <View style={{ width: 45, height: 45 }}>
          <Image
            source={assets.person03}
            resizeMode="contain"
            style={{ width: "100%", height: "100%" }}
          />
        </View>
      </View>

      {/* –ö–æ–Ω—Ç–µ–π–Ω–µ—Ä –¥–ª—è –ø–æ–ª—è –ø–æ–∏—Å–∫–∞ */}
      <View style={{ marginTop: SIZES.font }}>
        <View
          style={{
            width: "100%",
            borderRadius: SIZES.font,
            backgroundColor: COLORS.gray,
            flexDirection: "row",
            alignItems: "center",
            paddingHorizontal: SIZES.font,
            color: COLORS.white,
            paddingVertical: SIZES.small - 2,
          }}
        >
          <Image
            source={assets.search}
            resizeMode="contain"
            style={{ width: 20, height: 20, marginRight: SIZES.base }}
          />
          <TextInput
            placeholder="–ü–æ–∏—Å–∫ –ø–æ—Å—Ç–æ–≤"
            style={{ flex: 1 }}
            onChangeText={onSearch}
          />
        </View>
      </View>
    </View>
  );
};

export default HomeHeader;