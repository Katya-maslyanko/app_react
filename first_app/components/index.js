import HomeHeader from "./HomeHeader";
import Card from "./Card";
import { RectButton, CircleButton } from "./Button";
import { SubInfo, People, EndDate, Title } from "./SubInfo";
import FocusedStatusBar from "./FocusedStatusBar";
import DetailsDesc from './DetailsDesc';

export {
  HomeHeader,
  Card,
  SubInfo,
  People,
  EndDate,
  RectButton,
  CircleButton,
  Title,
  FocusedStatusBar,
  DetailsDesc,
};