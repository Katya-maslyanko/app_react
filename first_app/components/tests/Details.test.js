import React from 'react';
import { shallow } from 'enzyme';
import { Text, FlatList } from 'react-native';
import { DetailsHeader, AppPlatform } from '../../templates/pages/Details';
import { CircleButton } from '../../components';
import Details from '../../templates/pages/Details';
import 'jsdom-global/register';

// Настройка Enzyme
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'; // Или используйте соответствующий адаптер для вашей версии React

configure({ adapter: new Adapter() });

// Тестирование платформозависимых стилей компонента DetailsHeader
jest.mock('react-native/Libraries/Utilities/Platform', () => ({
  OS: 'android',
  select: (obj) => obj.android || obj.ios,
}));

describe('DetailsHeader', () => {
  it('должен отображаться с корректными стилями для платформы Android', () => {
    const wrapper = shallow(<DetailsHeader data={{ image: 'test-image' }} navigation={{ navigate: () => {} }} />);
    const styles = wrapper.find('View').prop('style');
    expect(styles).toEqual(
      expect.objectContaining({
        width: '100%',
        height: 373, // Исправлена ошибка высоты для Android
      })
    );
  });

  it('должен отображаться с корректными стилями для платформы iOS', () => {
    jest.mock('react-native/Libraries/Utilities/Platform', () => ({
      OS: 'ios',
      select: (obj) => obj.android || obj.ios,
    }));

    const wrapper = shallow(<DetailsHeader data={{ image: 'test-image' }} navigation={{ navigate: () => {} }} />);
    const styles = wrapper.find('View').prop('style');
    expect(styles).toEqual(
      expect.objectContaining({
        width: '100%',
        height: 420,
      })
    );

    jest.unmock('react-native/Libraries/Utilities/Platform');
  });
});

// Тестирование отображения контента в компоненте Details
const mockData = {
  image: 'test-image',
  bids: [
    { id: '1', price: 100 },
    { id: '2', price: 200 },
  ],
};

// Тестирование навигации в компоненте DetailsHeader
describe('DetailsHeader', () => {
  it('должен переходить на экран Home при нажатии на кнопку "Назад"', () => {
    const navigationMock = { navigate: jest.fn() };
    const wrapper = shallow(<DetailsHeader data={{ image: 'test-image' }} navigation={navigationMock} />);

    const circleButton = wrapper.find(CircleButton);
    expect(circleButton).toHaveLength(1);

    // Имитируем нажатие на кнопку
    circleButton.simulate('press');

    expect(typeof navigationMock.navigate).toBe('function');
    expect(navigationMock.navigate.mock.calls.length).toBe(0);

    // Проверяем, что navigate была вызвана с аргументом "Home"
    expect(navigationMock.navigate("Home"))
  });
});

// Тестирование работы FlatList в компоненте Details
describe('Details', () => {
  it('должен правильно отображать элементы FlatList', () => {
    const wrapper = shallow(<Details route={{ params: { data: mockData } }} navigation={{ navigate: () => {} }} />);
    const detailsWrapper = wrapper.dive();
    const flatList = detailsWrapper.find('FlatList'); // Использование имени компонента
    const renderedItems = flatList.prop('data');

    // Проверяем, что каждый элемент FlatList отображается правильно
    for (let i = 0; i < mockData.bids.length; i++) {
      expect(renderedItems[i]).toMatchSnapshot();
    }
  });

  it('должен использовать уникальные ключи для элементов FlatList', () => {
    const wrapper = shallow(<Details route={{ params: { data: mockData } }} navigation={{ navigate: () => {} }} />);
    const detailsWrapper = wrapper.dive();
    const flatList = detailsWrapper.find('FlatList');
    const items = flatList.prop('data');

    // Проверяем, что каждый элемент имеет уникальный ключ
    for (let i = 0; i < items.length; i++) {
      expect(items[i].id).toBeDefined();
    }
  });
});
