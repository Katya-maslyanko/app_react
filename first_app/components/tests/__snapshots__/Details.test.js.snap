// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details должен правильно отображать элементы FlatList 1`] = `
{
  "id": "1",
  "price": 100,
}
`;

exports[`Details должен правильно отображать элементы FlatList 2`] = `
{
  "id": "2",
  "price": 200,
}
`;
